tests/test_runner.py

```diff
@@ -248,3 +255,36 @@ async def test_worker_long_chain_jobs():
     for job_id in job_ids:
         state = await backend.get_job_state("runner", job_id)
         assert state == State.COMPLETED
+
+
+async def test_worker_logs_exceptions():
+    def _setup_custom_logging():
+        class TestHandler(logging.Handler):
+            def emit(self, record):
+                logs.append(record)
+
+        logger = logging.getLogger('asyncmq')
+        handler = TestHandler()
+        logger.addHandler(handler)
+        logger.setLevel(logging.DEBUG)
+
+    async def run_test():
+        async with anyio.create_task_group() as tg:
+            tg.start_soon(run_worker, "runner", backend)
+            await anyio.sleep(0.5)
+            tg.cancel_scope.cancel()
+
+    logs = []
+    _setup_custom_logging()
```

Comment:
It seems `caplog` works fine with trio but does not capture any message for the asyncio backend, thus the strange logging setup.